static/app/actionCreators/pageFilters.tsx

```diff
@@ -299,17 +313,34 @@ function updateParams(obj: PageFiltersUpdate, router?: Router, options?: Options
     return;
   }
 
-  if (options?.save) {
-    const {organization} = OrganizationStore.getState();
-    const {selection} = PageFiltersStore.getState();
-    const orgSlug = organization?.slug ?? null;
+  const routerAction = options?.replace ? router.replace : router.push;
+
+  routerAction({pathname: router.location.pathname, query: newQuery});
+}
 
-    setPageFiltersStorage(orgSlug, selection, newQuery);
+/**
+ * Save the current page filters to local storage
+ */
+async function persistPageFilters(options?: Options) {
+  if (!options?.save) {
+    return;
   }
 
-  const routerAction = options?.replace ? router.replace : router.push;
+  // XXX(epurkhiser): Since this is called immediately after updating the
+  // store, wait for a tick since stores are not updated fully synchronously..
```

Comment:
```suggestion
  // store, wait for a tick since stores are not updated fully synchronously.
```

Reply:
will clean up these minor nits in follow up 👍